rcl_lifecycle/src/default_state_machine.c

```diff
@@ -659,6 +660,9 @@ rcl_lifecycle_init_default_state_machine(
   rcl_ret_t ret = RCL_RET_ERROR;
   // Used for concatenating error messages in the fail: block.
   const char * fail_error_message = "";
+  char * init_error_string = "";
+  char * fini_error_string = "";
```

Comment:
consider renaming this to `init_error_message` and `fini_error_message` to stay consistent with `fail_error_message`. 
Also I would recommend to add a bit of a docstring on top of each of the three error messages to get an idea on what's going on here.

Reply:
Also, why do we need 3 strings here?
I would assume 2 should be sufficient. 

* init_error_message: The original cause or error why we land in `fail:`
* fini_error_message: A conglomerate of failures happening within `fail:`.